src/components/HomePage/SearchResultsArea/SearchResultsWidget/useOffersSearcher.js

```diff
@@ -0,0 +1,95 @@
+import { useCallback, useState, useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import {
+    resetOffersFetchError,
+    setLoadingOffers,
+    setOffersFetchError,
+    setSearchOffers,
+    setSearchQueryToken,
+} from "../../../../actions/searchOffersActions";
+import {
+    searchOffers,
+    BadResponseException,
+    NetworkFailureException,
+} from "../../../../services/offerService";
+
+export default (filters) => {
+
+    const dispatch = useDispatch();
+    const searchQueryToken = useSelector((state) => state.offerSearch.queryToken);
+
+    const [hasMoreOffers, setHasMoreOffers] = useState(true);
+    const [moreOffersFetchError, setMoreOffersFetchError] = useState(null);
+    const [moreOffersLoading, setMoreOffersLoading] = useState(false);
+
+    // The "search" and "loadMoreOffers" functions do not share the same state;
+    // When we run "setHasMoreOffers(false)" on the "loadMoreOffers" function,
+    // the "search" function does not know that the "hasMoreOffers" variable has changed;
+    // In the same way, when we run "setHasMoreOffers(true) on the "search" function,
+    // the "loadMoreOffers" function does not know that the "hasMoreOffers" variable has changed;
+    // Then, when the "loadMoreOffers" function is executed after a previous execution where the
+    // "hasMoreOffers" variable became "false", the state of this variable is still false, which
+    // prevents the fetching of new offers.
+    // Knowing that, I needed a way to set the "hasMoreOffers" variable to "true" when the previous fact
+    // happens: setting the "hasMoreOffers" to true when the "queryToken" (which is stored in redux) changes
+    useEffect(() => {
+        setHasMoreOffers(true);
+    }, [searchQueryToken]);
```

Comment:
@imnotteixeira I found out that we have a bug when we fetch all the offers of a search and then do another search. What do you think of this way to solve that?

Reply:
I have a solution, but I'm not sure you're going to like it...

Maybe it's time to go full-hook and ditch redux for search functionality.

The main idea is: the HomePage Controller (might need to create it) will start storing the search state by calling the search hook. The search hook will be modified to hold the state and export the setters of each filter and the search value etc. Then it will expose the `search` and `loadMore` functions, as it does currently. It will additionally hold the results array, and errors and loading state, etc... like the redux state currently has.

Then SearchArea and SearchResultsWidget components can both access the HomePage context which will have these methods/data accessible, and most importantly **common** to both. There can only be 1 search at any given time, which is what we want. Components that currently use the redux state for search can easily call `useContext(HomePageContext)` and retrieve whatever they need, no need to pass props around, no connect components to redux.

I'm not sure if we will only ever have the search on the homepage. Maybe some day we will want to keep the search state while the student goes through other pages (viewing offers, adding them to his personal track-list, dunno) and keep the results in a sort-of cart that they can quickly get back to. But for now this is good enough, no need to handle the context above for it to be accessed in other pages.

Reply:
I think it's a good idea

Reply:
@imnotteixeira That refactoring will take some time, mainly due to all the changes in the tests. What do you think of leaving this as it is now (because it works) and opening an issue for that?

Reply:
> @imnotteixeira I found out that we have a bug when we fetch all the offers of a search and then do another search. What do you think of this way to solve that?

Then is this a bug or not?

In any case, we should migrate in the future. Search is the next big system still not using hooks/context. Migrating will probably make it simpler. Create the issue and link it here please

Reply:
It was a bug, but I solved it using that `useEffect`, which uses the `searchQueryToken` from `redux`.